lib/Conversion/TorchToLinalg/Reduction.cpp

```diff
@@ -322,6 +389,46 @@ class ConvertReductionOp : public ConversionPattern {
     auto resultType = getTypeConverter()
                           ->convertType(op->getResult(0).getType())
                           .cast<RankedTensorType>();
+
+    if (auto normOp = dyn_cast<AtenLinalgVectorNormOp>(op)) {
+      auto elemType = resultType.getElementType();
+
+      if (!elemType.isa<mlir::FloatType>())
+        return rewriter.notifyMatchFailure(
+            op, "only float types are valid for vector norm ops");
+
+      // Cast `ord` to float so that we can readily pass it math.powf.
+      auto ordValue = operands[1];
+      if (ordValue.getType().isa<mlir::IntegerType>())
+        ordValue = rewriter.create<arith::SIToFPOp>(loc, elemType, ordValue);
+
+      // Sum each element of the tensor after computing the exponentiation.
+      auto sumOp =
+          createNormSumReduction(loc, elemType, opInfo, ordValue, rewriter);
+      if (!sumOp)
+        return failure();
+
+      // TODO: Add support for L0 norm.
+      auto epsilon = 1e-5;
+      auto ordLiteral = 0.0;
+      if (matchPattern(ordValue, m_TorchConstantFloat(&ordLiteral)) &&
+          fabs(ordLiteral) < epsilon)
+        return rewriter.notifyMatchFailure(op, "unimplemented: L0 norm");
+
+      // Raise each summed value to the inverse of the order of the norm.
+      auto oneAttr = rewriter.getFloatAttr(elemType, 1.0);
+      auto oneValue = rewriter.create<arith::ConstantOp>(loc, oneAttr);
+      auto inverseOrdValue =
+          rewriter.create<arith::DivFOp>(loc, oneValue, ordValue);
+      auto expOp = createNormExp(loc, elemType, inverseOrdValue, sumOp, opInfo,
+                                 rewriter);
+      if (!expOp)
+        return failure();
+
+      rewriter.replaceOpWithNewOp<tensor::CastOp>(op, resultType, expOp);
+      return success();
+    }
+
     Value initElem = createLinalgNeutralElementForReduceOp(
```

Comment:
It seems to me that the main difference between the norm op and the other reduction ops is the exponent after the reduction has taken place. I think we should try to have the norm op also share the code below and only add an addendum to it with the exponent calculation so that all reduction ops share most of the logic and have similar structure

Reply:
Excellent point.  Fixed in commit ee3549b, which is a somewhat long patch, but I can add more comments if it gets too difficult to review.